deposits.md

```diff
@@ -0,0 +1,225 @@
+# Deposits
```

Comment:
Its definitely the cleanest way to handle deposit transactions as a new transaction type but will require a lot of work getting them added to all of the libraries. It seems like we will need to write the serialization in go (hopefully get it into geth, will need to deserialize from the engine RPC), in JS (hopefully get into ethers, required for fetching transactions) and would be nice to get it in Rust (ethers-rs)